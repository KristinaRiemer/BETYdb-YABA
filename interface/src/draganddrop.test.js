import React from "react";
import { shallow } from 'enzyme';
import DragandDrop from "./components/draganddrop"
import Dropzone from 'react-dropzone'
import { fireEvent, render } from '@testing-library/react'


describe('Test Dropzone component', () => {
  it("should accept files", async() => {
    const file = new File([
      JSON.stringify({ping: true})
    ], 'ping.json', { type: 'application/json' })
    const data = mockData([file])
    const onDragEnter = jest.fn()
  
    const ui = (
      <Dropzone onDragEnter={onDragEnter}>
        {({ getRootProps, getInputProps }) => (
          <div {...getRootProps()}>
            <input {...getInputProps()} />
          </div>
        )}
      </Dropzone>
    )
    const { container } = render(ui)
    const dropzone = container.querySelector('div')
  
    dispatchEvt(dropzone, 'dragenter', data)
    await flushPromises(ui, container)
  
    expect(onDragEnter).toHaveBeenCalled()
  });
  
  it('should render correctly with no props', () => {
    const component = shallow(<DragandDrop/>);
    
    expect(component).toMatchSnapshot();
  });  
});

function flushPromises(ui, container) {
  return new Promise(resolve =>
    setImmediate(() => {
      render(ui, { container })
      resolve(container)
    })
  )
}

function dispatchEvt(node, type, data) {
  const event = new Event(type, { bubbles: true })
  Object.assign(event, data)
  fireEvent(node, event)
}

function mockData(files) {
  return {
    dataTransfer: {
      files,
      items: files.map(file => ({
        kind: 'file',
        type: file.type,
        getAsFile: () => file
      })),
      types: ['Files']
    }
  }
}
