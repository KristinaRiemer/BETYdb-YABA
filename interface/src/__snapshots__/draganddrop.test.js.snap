// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dropzone component should render correctly with no props 1`] = `
<div
  className="container"
>
  <div
    className="dropzone"
    onBlur={null}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={null}
    onKeyDown={null}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fafafa",
        "borderColor": "#969899",
        "borderRadius": 5,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "color": "#555",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "outline": "none",
        "padding": "20px",
        "transition": "border .24s ease-in-out",
      }
    }
  >
    <input
      accept=""
      autoComplete="off"
      multiple={true}
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
    <div
      className="browse"
    >
      <button
        className="primary ripple-primary"
        onClick={[Function]}
      >
        BROWSE
      </button>
      <span>
        or drop files here
      </span>
    </div>
    <aside />
  </div>
</div>
`;
